### ISSUE TITLE
[a11y]: The Link component and others don't comply with their accessibility guidelines

### ISSUE THREAD

**Issue by adamalston** [carbon-design-system/carbon#18607]
### Package

@carbon/react

### Package version

1.76.0

### Automated testing tool and ruleset

N/A

### Description

https://carbondesignsystem.com/components/link/accessibility/

> Links are reached by Tab key and activated by Space and Enter keys.

The `Link` is just a wrapper for the `a` element. The MDN docs (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) for that say:

> If the `href` attribute is present, pressing the enter key while focused on the `<a>` element will activate it.

There are a few questions here:

1. Should the `Link` component have native handling for the space key?
   1. If so, why does Carbon have this additional requirement for the space key to activate the link that goes beyond all other links (`a` elements) on the web?
   2. If not, is every usage of a `Link` by a consumer of Carbon supposed to have an additional `onKeyDown` event handler added to it?

This issue doesn't just apply to the `Link` component. There are other Carbon components (e.g., `SideNavLink`, `SideNavMenuItem`, etc.) that wrap `a` elements whose accessibility guidelines say that the space key should be used to activate them. However, the Storybook entries (e.g., https://react.carbondesignsystem.com/?path=/story/components-ui-shell-sidenav--fixed-side-nav) that include these elements do not follow those guidelines.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-link--default

### Steps to reproduce

1. Go to the Storybook.
2. Focus the link.
3. Try activating the link using the Space key.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Interesting, thanks for bringing this up! My view is it's likely the guidance on the website is wrong and/or outdated, especially when reading through https://github.com/w3c/aria/issues/1701

@mbgower Should `a` tags in the system respond to `space`? If no, maybe we could update the docs to state that they're commonly activated with <kbd>enter</kbd> but we leave it up to UA's which might also allow activation via <kbd>space</kbd>. What do you think?

**Comment by mbgower** [comment]
> Maybe it would be easier to clarify which components, if any, should be able to be activated with the space key.

I don't understand what the issue is. Carbon follows established keyboard operation for native html components, which are typically foundational in even the most complex widgets. Where a button is used in any such component (such as the ones you cite) both Space and Enter normally work. The same is the case with uses of checkboxes, etc.,  inside a complex component.
Beyond the error you identified with the link documentation, what is the issue?

Here is a [diagram](https://www.ibm.com/able/images/keyboard.png) of basic keyboard interaction

**Comment by adamalston** [comment]
Apologies, I think I misread https://carbondesignsystem.com/components/UI-shell-left-panel/accessibility/#keyboard-interaction. I will keep https://www.ibm.com/able/images/keyboard.png in mind as I'm working through other accessibility issues that have been reported. Thanks.
